pubtools/pulplib/_impl/client/client.py

```diff
@@ -179,6 +182,42 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def upload_one_file(
+        self,
+        file_name,
+        repo_id,
+        type_id,
+        unit_key,
+        unit_metadata=None,
+        override_config=None,
+    ):
+        """Including four steps uploading a file:
+            1. request upload id from pulp
+            2. upload contents to pulp
+            3. import the uploaded content to wanted repo
+            4. delete the reuqest id
+        """
+        upload_id = self._request_upload().result()["upload_id"]
+        offset = 0
+        with open(file_name) as f:
```

Comment:
I think this is broken on Python 3, isn't it going to open the file as text and try decoding it using utf8 (or user's locale)?  Just tried it locally and it seems to be so.  I think you'll need to open with mode 'b' to read the binary data.

What I tried locally:

```
>>> open('tmp')
<_io.TextIOWrapper name='tmp' mode='r' encoding='UTF-8'>
>>> open('tmp', 'rb')
<_io.BufferedReader name='tmp'>
```

The first case will certainly crash or do the wrong thing when trying to upload e.g. an ISO or RPM file.